nik4.py

```diff
@@ -18,16 +18,19 @@
 TILE_BUFFER = 128
 IM_MONTAGE = 'montage'
 
-p3857 = mapnik.Projection('+proj=merc +a=6378137 +b=6378137 +lat_ts=0.0 +lon_0=0.0 +x_0=0.0 +y_0=0 +k=1.0 +units=m +nadgrids=@null +no_defs +over')
-p4326 = mapnik.Projection('+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs')
-transform = mapnik.ProjTransform(p4326, p3857)
+proj_target = mapnik.Projection('+proj=merc +a=6378137 +b=6378137 +lat_ts=0.0 +lon_0=0.0 +x_0=0.0 +y_0=0 +k=1.0 +units=m +nadgrids=@null +no_defs +over')
+proj_lonlat = mapnik.Projection('+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs')
+proj_web_merc = mapnik.Projection('+proj=merc +a=6378137 +b=6378137 +lat_ts=0.0 +lon_0=0.0 +x_0=0.0 +y_0=0 +k=1.0 +units=m +nadgrids=@null +no_defs +over')
+transform = mapnik.ProjTransform(proj_lonlat, proj_target)
+transform_lonlat_webmerc = mapnik.ProjTransform(proj_lonlat, proj_web_merc)
 
 def layer_bbox(m, names, bbox=None):
 	"""Calculate extent of given layers and bbox"""
 	for layer in (l for l in m.layers if l.name in names):
 		# it may as well be a GPX layer in WGS84
-		p = mapnik.Projection(layer.srs)
-		lbbox = layer.envelope().inverse(p).forward(p3857)
+		layer_proj = mapnik.Projection(layer.srs)
+		box_trans = mapnik.ProjTransform(layer_proj, proj_target)
+		lbbox = box_trans.forward(mapnik.Box2d.from_string('995585 6291591 998380 6293955'))
```

Comment:
What do these numbers mean? You are not using `layer.envelope()` now, as far as I see.

Reply:
Oh yeah, my stupidity. I fixed it.